src/components/common/ColonyActions/ColonyActions.tsx

```diff
@@ -33,18 +33,31 @@ const MSG = defineMessages({
 //   ethDomainId?: number;
 // };
 
+const ITEMS_PER_PAGE = 10;
+
 const ColonyActions = (/* { ethDomainId }: Props */) => {
   const { colony } = useColonyContext();
 
-  const { data, loading: loadingActions } = useGetColonyActionsQuery({
+  const [sortDirection, setSortDirection] = useState<SortDirection>(
```

Comment:
This all could perhaps live in a hook? Just to try and keep the component as neat as possible.

Reply:
I extracted the entire fetching/sorting/pagination logic into its own hook, `usePaginatedActions`. I'm not sure if this was your suggestions, or maybe you wanted me to extract just the sorting/pagination logic so it's reusable, but having thought about it, sorting and pagination will not always go hand in hand. I'll revisit it once there's another use case for sorting or pagination